core/src/css/themes/high-contrast-light.scss

```diff
@@ -89,12 +100,48 @@ $colors:  (
       --ion-color-#{$color-name}-tint: #{map.get($value, tint)};
     }
 
+    --ion-background-color: #{$background-color};
+    --ion-background-color-rgb: #{color-to-rgb-list($background-color)};
+
+    --ion-text-color: #{$text-color};
+    --ion-text-color-rgb: #{color-to-rgb-list($text-color)};
+
     --ion-placeholder-opacity: 0.8;
+
+    /// Only the item borders should increase in contrast
+    /// Borders for elements like toolbars should remain the same
+    --ion-item-border-color: #7a7a7a;
+
+    --ion-text-color-step-50: #{mix($lightest-text-color, $darkest-text-color, 5%)};
+    --ion-text-color-step-100: #{mix($lightest-text-color, $darkest-text-color, 10%)};
+    --ion-text-color-step-150: #{mix($lightest-text-color, $darkest-text-color, 15%)};
+    --ion-text-color-step-200: #{mix($lightest-text-color, $darkest-text-color, 20%)};
+    --ion-text-color-step-250: #{mix($lightest-text-color, $darkest-text-color, 25%)};
+    --ion-text-color-step-300: #{mix($lightest-text-color, $darkest-text-color, 30%)};
+    --ion-text-color-step-350: #{mix($lightest-text-color, $darkest-text-color, 35%)};
+    --ion-text-color-step-400: #{mix($lightest-text-color, $darkest-text-color, 40%)};
+    --ion-text-color-step-450: #{mix($lightest-text-color, $darkest-text-color, 45%)};
+    --ion-text-color-step-500: #{mix($lightest-text-color, $darkest-text-color, 50%)};
+    --ion-text-color-step-550: #{mix($lightest-text-color, $darkest-text-color, 55%)};
+    --ion-text-color-step-600: #{mix($lightest-text-color, $darkest-text-color, 60%)};
+    --ion-text-color-step-650: #{mix($lightest-text-color, $darkest-text-color, 65%)};
+    --ion-text-color-step-700: #{mix($lightest-text-color, $darkest-text-color, 70%)};
+    --ion-text-color-step-750: #{mix($lightest-text-color, $darkest-text-color, 75%)};
+    --ion-text-color-step-800: #{mix($lightest-text-color, $darkest-text-color, 80%)};
+    --ion-text-color-step-850: #{mix($lightest-text-color, $darkest-text-color, 85%)};
+    --ion-text-color-step-900: #{mix($lightest-text-color, $darkest-text-color, 90%)};
+    --ion-text-color-step-950: #{mix($lightest-text-color, $darkest-text-color, 95%)};
   }
+}
 
+@mixin high-contrast-light-md-theme() {
   // Toast
   // --------------------------------------------------
 
+  & ion-toast {
+    --color: var(--ion-background-color);
+  }
+
```

Comment:
One thing I missed during the design phase of this toast workaround is that the text stepped colors would also be changing. I built this solution before I worked on the new text stepped colors architecture.

The problem is that `--color` by default uses a stepped color. But since those stepped colors are changing in high contrast, the way they mix using `color-mix` also changes. This resulted in text not having sufficient contrast. As a result, all text in high contrast mode will use the background color of the page.

Note that this is not a type. The MD toast background is a dark gray in light mode, so using the page background color (white) will ensure correct contrast. Similarly, the background is white in dark mode (where the page background is dark gray).

Reply:
I had to make the same adjustments in the high contrast dark theme too